src/decide.ts

```diff
@@ -1,52 +1,66 @@
 /* eslint-disable no-param-reassign */
 
-import { CacheMetadata, CacheMetadataKey, CacheMetadataRepository } from './cache-metadata';
+import { CacheMetadataKey, CacheMetadataRepository } from './cache-metadata';
 import Config from './config';
 import { service } from './dynamodb';
 import { FileHasher } from './hash';
 import { count } from './util';
 
 /**
  * If a config has changes, its steps are merged into the final build. Otherwise, it is excluded, and its excluded_steps
- * are merged in instead. There are exceptions:
+ * are merged in instead.
+ */
+function updateDecisionsForChanges(configs: Config[]): void {
+  configs.forEach((config) => {
+    if (config.changes.length > 0) {
+      config.decide(true, `${count(config.changes, 'matching change')}: ${config.changes.join(', ')}`);
+    }
+  });
+}
+
+/**
+ * Provides for manually supplied overrides to the inclusion decisions:
  *  - An env var named PIPELINE_RUN_ALL, set to any value, indicates that all steps should run
  *  - An env var named PIPELINE_RUN_<NAME>, where NAME is the UPPER_SNAKE_CASE version of the component pipeline name,
- *    set to any value, indicates that step should run
+ *    set to any value, indicates that step should run - and the opposite for PIPELINE_NO_RUN_<NAME>
+ *  - An env var named PIPELINE_RUN_ONLY, with value of a single pipeline name, indicates only that step should run
  */
-export function updateDecision(config: Config): void {
-  if (process.env.PIPELINE_RUN_ALL) {
-    config.decide(true, 'been forced to by PIPELINE_RUN_ALL');
-    return;
-  }
-
-  const envVarName = config.monorepo.name.toLocaleUpperCase().replace(/-/g, '_');
+function updateDecisionsForEnvVars(configs: Config[]): void {
+  configs.forEach((config) => {
+    if (process.env.PIPELINE_RUN_ALL) {
+      config.decide(true, 'been forced to by PIPELINE_RUN_ALL');
+      return;
+    }
 
-  const overrideExcludeKey = `PIPELINE_NO_RUN_${envVarName}`;
-  if (process.env[overrideExcludeKey]) {
-    config.decide(false, `been forced NOT to by ${overrideExcludeKey}`);
-    return;
-  }
+    const envVarName = config.monorepo.name.toLocaleUpperCase().replace(/-/g, '_');
 
-  const overrideIncludeKey = `PIPELINE_RUN_${envVarName}`;
-  if (process.env[overrideIncludeKey]) {
-    config.decide(true, `been forced to by ${overrideIncludeKey}`);
-    return;
-  }
+    const overrideExcludeKey = `PIPELINE_NO_RUN_${envVarName}`;
+    if (process.env[overrideExcludeKey]) {
+      config.decide(false, `been forced NOT to by ${overrideExcludeKey}`);
+      return;
+    }
```

Comment:
Put this above the `PIPELINE_RUN_ALL` check as it would be good to be able to combine them with:
```
PIPELINE_RUN_ALL=true
PIPELINE_NO_RUN_FOO=false
```
which would run all except foo.